Address Samir's OPDIR review to Refine service provisioning details and nits corrections (#41)
ietf-ivy-wg/network-inventory-topology: draft-ietf-ivy-network-inventory-topology.md

--- abstract

   This document defines a YANG data model that extends the network

> topology model RFC 8345 to represent a physical network

Reviewer: ```suggestion
topology model (RFC 8345) to represent a physical network
```

> that forms an underlay for logical networks above it.  The module

Reviewer: ```suggestion
that forms an underlay for logical networks above it.  The data model
```

## Determine Available Resources of Service Attachment Points (SAPs)

The inventory topology data model correlates underlay physical
resource information with the SAP network model {{!RFC9408}}.

> While the SAP model provides the provider network view with the

Reviewer: ```suggestion
While the SAP data model provides the provider network view with the
```

points from which services can be attached, the inventory

> topology model maps those SAPs to their underlying physical

Reviewer: ```suggestion
topology data model maps those SAPs to their underlying physical
```

ports, enabling the orchestrator to verify whether a candidate
SAP has sufficient physical capacity.

{{nwi-topology-usage}} illustrates the query interactions.
During service provisioning, the orchestrator queries the SAP

> model (e.g., obtaining a list of SAPs across multiple PE nodes

Reviewer: ```suggestion
data model (e.g., obtaining a list of SAPs across multiple PEs
```

as shown in Appendix A of {{!RFC9408}}), and then uses the

> inventory topology model to check the physical resources of the

Reviewer: ```suggestion
inventory topology data model to check the physical resources of the
```

candidate SAPs.  Specifically, the "parent-termination-point"
of a SAP is mapped to the corresponding "port-component-ref"
in the inventory topology, allowing the orchestrator to verify
port availability and capacity.

If the physical port underlying a candidate SAP has insufficient
resources (e.g., port speed fully utilized), the orchestrator

> can select an alternative SAP that maps to a different port

Reviewer: ```suggestion
can select an alternate SAP that maps to a different port
```

with adequate capacity.  If no alternative SAP is available,
the orchestrator flags the request for manual intervention,
providing the operator with precise inventory information
about the bottleneck (e.g., "Port GE0/6/1 on NE-PE1 is at 95% utilization").
The resource constraint can also feed into a "what-if" analysis
(see {{sec-whatif}}) to evaluate hardware upgrades or
alternative underlay paths.

~~~~ aasvg
                     +-----------------+
                     |     Customer    |
                     +--------+--------+
     Customer Service request |

>         (e.g., L3SM, L2SM)    |

Reviewer: ```suggestion
        (e.g., L3SM and L2SM)  |
```

                     +--------v--------+
                     |    Service      |
                     |  Orchestration  |
                     +------+---+------+
            (1a) Query SAPs |   | (1b) Verify physical

>             via SAP Model   |   | capacity via Inventory Topology

Reviewer: ```suggestion
        via SAP data model   |   | capacity via Inventory Topology
```

{::include-fold ./yang/examples/link-type-example.json}
~~~~

> # JSON Example of an MPO (Multi-fibre Push On) Breakout-Channel Port

Reviewer: ```suggestion
# JSON Example of an Multi-fibre Push On (MPO) Breakout-Channel Port
```
